### PR TITLE
Add docstring for get_tokens_of_type().

### PR DESCRIPTION
**Pull request by obi1kenobi**

### CODE REVIEW

File: graphql_compiler/interpreter/typedefs.py [hunk 1]
```diff
@@ -156,8 +154,72 @@ def get_tokens_of_type(
...
+        type named Foo, queries must start by querying for Foo in order to be valid for the schema:
+        {
+            Foo {
+                < stuff here >
+            }
+        }
+
+        To compute the results for such a query, the interpreter would call get_tokens_of_type()
+        with "Foo" as the type_name value. As get_tokens_of_type() yields tokens, the interpreter
+        uses those tokens to perform the rest of the query via the remaining API methods.
```
Comment (branen) [inline graphql_compiler/interpreter/typedefs.py:192]:
```suggestion
        uses those tokens to perform the rest of the query via the remaining interpreter API methods.
```
If the reader's thinking about how to implement an adapter over some other API, the clarification may help here.

File: graphql_compiler/interpreter/typedefs.py [hunk 2]
```diff
@@ -156,8 +154,72 @@ def get_tokens_of_type(
...
+        any interpreted query. However, due to the generator-style operation of the interpreter,
+        the call to get_tokens_of_type() is *not* guaranteed to be the first call across the four
+        methods that comprise this API -- one or more calls to the other methods may precede it.
+
+        Args:
+            type_name: name of the vertex type for which to yield tokens. Guaranteed to be:
+                       - the name of a type defined in the schema being queried, and specifically
+                       - one of the types defined in the schema's root query type:
+                         http://spec.graphql.org/June2018/#sec-Root-Operation-Types
+            runtime_arg_hints: values of any runtime arguments provided to the query for use in
```
Comment (branen) [inline graphql_compiler/interpreter/typedefs.py:204]:
```suggestion
            runtime_arg_hints: names and values of any runtime arguments provided to the query for use in
```

**Review by branen — CHANGES_REQUESTED** [review]
Looks good!  I made a few suggestions that I think will make it slightly easier to read for naive users.

**Comment by obi1kenobi** [comment]
Thanks!

File: graphql_compiler/interpreter/typedefs.py [hunk 3]
```diff
@@ -156,8 +154,73 @@ def get_tokens_of_type(
...
+        methods that comprise this API -- one or more calls to the other methods may precede it.
+
+        Args:
+            type_name: name of the vertex type for which to yield tokens. Guaranteed to be:
+                       - the name of a type defined in the schema being queried, and specifically
+                       - one of the types defined in the schema's root query type:
+                         http://spec.graphql.org/June2018/#sec-Root-Operation-Types
+            runtime_arg_hints: names and values of any runtime arguments provided to the query
+                               for use in filtering operations (e.g. "$foo").
+            used_property_hints: the property names of the requested vertices the that
```
Comment (chewselene) [inline graphql_compiler/interpreter/typedefs.py:207]: typo: extra `the` (before the `that`)

File: graphql_compiler/interpreter/typedefs.py [hunk 4]
```diff
@@ -156,8 +154,73 @@ def get_tokens_of_type(
...
+        the call to get_tokens_of_type() is *not* guaranteed to be the first call across the four
+        methods that comprise this API -- one or more calls to the other methods may precede it.
+
+        Args:
+            type_name: name of the vertex type for which to yield tokens. Guaranteed to be:
+                       - the name of a type defined in the schema being queried, and specifically
+                       - one of the types defined in the schema's root query type:
+                         http://spec.graphql.org/June2018/#sec-Root-Operation-Types
+            runtime_arg_hints: names and values of any runtime arguments provided to the query
+                               for use in filtering operations (e.g. "$foo").
```
[the remark this answers is not in the document]
Comment (obi1kenobi) [reply graphql_compiler/interpreter/typedefs.py:206]: Good point. Replaced with `$arg_name` everywhere to avoid this problem in other places as well.

**Review by branen — APPROVED** [review]

**Pull request merged**